src/main/java/de/hysky/skyblocker/skyblock/dungeon/puzzle/boulder/Boulder.java

```diff
@@ -0,0 +1,217 @@
+     *
+     * @param world The client world.
+     * @param x     The x-coordinate of the block.
+     * @param y     The y-coordinate of the block.
+     * @param z     The z-coordinate of the block.
+     * @return The type of block at the specified position.
+     */
+    public static String getBlockType(ClientWorld world, int x, int y, int z) {
+        Block block = world.getBlockState(DungeonManager.getCurrentRoom().relativeToActual(new BlockPos(x, y, z))).getBlock();
+        return (block == Blocks.BIRCH_PLANKS || block == Blocks.JUNGLE_PLANKS) ? "B" : ".";
+    }
+
+    /**
+     * Checks for blocks along the line between two points in the world.
+     * Returns the position of a block if it found a button on the line, if any.
+     *
+     * @param world   The client world.
+     * @param point1  The starting point of the line.
+     * @param point2  The ending point of the line.
+     * @return The position of the block found on the line, or null if no block is found.
+     */
+    private static BlockPos checkForButtonBlocksOnLine(ClientWorld world, Vec3d point1, Vec3d point2) {
+        double x1 = point1.getX();
+        double y1 = point1.getY() + 1;
+        double z1 = point1.getZ();
+
+        double x2 = point2.getX();
+        double y2 = point2.getY() + 1;
+        double z2 = point2.getZ();
+
+        int steps = (int) Math.max(Math.abs(x2 - x1), Math.max(Math.abs(y2 - y1), Math.abs(z2 - z1)));
+
+        double xStep = (x2 - x1) / steps;
+        double yStep = (y2 - y1) / steps;
+        double zStep = (z2 - z1) / steps;
+
+
+        for (int step = 0; step <= steps; step++) {
+            double currentX = x1 + step * xStep;
+            double currentY = y1 + step * yStep;
+            double currentZ = z1 + step * zStep;
+
+            BlockPos blockPos = BlockPos.ofFloored(currentX, currentY, currentZ);
+            Block block = world.getBlockState(blockPos).getBlock();
+
+            if (block == Blocks.STONE_BUTTON) {
+                return blockPos;
+            }
+
+        }
+        return null;
+    }
+
+    /**
+     * Retrieves the bounding box of a block in the world.
+     *
+     * @param world The client world.
+     * @param pos   The position of the block.
+     * @return The bounding box of the block.
+     */
+    public static Box getBlockBoundingBox(BlockView world, BlockPos pos) {
+        BlockState blockState = world.getBlockState(pos);
+        return blockState.getOutlineShape(world, pos).getBoundingBox().offset(pos);
+    }
+
+    @Override
+    public void render(WorldRenderContext context) {
+        if (!shouldSolve() || !SkyblockerConfigManager.get().locations.dungeons.solveBoulder || !DungeonManager.isCurrentRoomMatched())
+            return;
+        float alpha = 1.0f;
+        float lineWidth = 5.0f;
+
+        if (linePoints != null && linePoints.length > 0) {
+            RenderHelper.renderLinesFromPoints(context, linePoints, ORANGE_COLOR_COMPONENTS, alpha, lineWidth, true);
```

Comment:
The lines need to be rendered one at a time for the line width and normals to not get messed up. (See the last part of `Waterboard#render`.)